### PR TITLE
Setting to make On This Day smart album public

### PR DESCRIPTION
**Pull request by aldjordje**

### CODE REVIEW

File: database/migrations/2023_01_09_133603_public_on_this_day.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	public function up(): void
+	{
+		DB::table('configs')->insert([
+			[
+				'key' => 'public_on_this_day',
+				'value' => '0',
+				'cat' => 'Smart Albums',
+				'type_range' => '0|1',
+				'confidentiality' => '0',
+				'description' => 'Make On This Day smart album accessible to anonymous users',
```
Comment (ildyria) [inline database/migrations/2023_01_09_133603_public_on_this_day.php]:
```suggestion
				'description' => 'Make "On This Day" smart album accessible to anonymous users',
```

**Review by ildyria — APPROVED** [review]
LGTM.

**Comment by ildyria** [comment]
Can you see if you can add some tests to see if the flag works ?

**Comment by aldjordje** [comment]
> Can you see if you can add some tests to see if the flag works ?

@ildyria , I separated tests in different commit. I didn't have better idea for tests because we are testing multiple things at the same time.

**Comment by ildyria** [comment]
@aldjordje I am waiting for #1709 to be merged into master. Then we can merge it into your branch and get those tests to pass. Sorry for the delay.

File: tests/Feature/SharingWithNonAdminUserAndNoPublicSearchTest.php [hunk 2]
```diff
@@ -142,7 +160,7 @@ public function testPublicUnsortedPhotoAndPhotoInSharedAlbum(): void
 				$this->generateExpectedPhotoJson(static::SAMPLE_FILE_MONGOLIA_IMAGE, $this->photoID2, $this->albumID1),
 			]
 		));
-		$responseForUnsorted->assertJsonMissing(['id' => $this->photoID2]);
+		$responseForRecent->assertJsonMissing(['id' => $this->photoID1]);
```
Comment (ildyria) [inline tests/Feature/SharingWithNonAdminUserAndNoPublicSearchTest.php:163]: nicely spotted!

**Review by ildyria — APPROVED** [review]
LGTM

**Pull request merged**